src/Agent.Worker/Build/BuildServer.cs

```diff
@@ -60,7 +60,7 @@ public BuildServer(VssConnection connection, Guid projectId)
                 },
             };
 
-            return await _buildHttpClient.UpdateBuildAsync(build, _projectId, buildId, cancellationToken: cancellationToken);
```

Comment:
do we have to remove this?

Reply:
The `UpdateBuildAsync()` overload with the `projectId` and `buildId` parameters passed in is now deprecated (I got a build warning alerting me of this fact).

The new overload reads `projectId`and `buildId` from the `build` object, which we're already doing in the code anyways, so there's no functional change.

Reply:
i worried about back compat, latest client talk to TFS2017, double check with Build Team, make sure their client is back compat.